text/0028-bill-of-materials.md

```diff
@@ -0,0 +1,105 @@
+# Implement a Bill of Materials Across Paketo
+
+## Summary
+
+A bill of materials (BOM) is a set of metadata that provides an account of the
+dependencies and tools that are both present in an application image, as well
+those used in the construction of that image. The availability of this data is
+a key value proposition of Cloud Native Buildpacks.
+
+As an implementation of Cloud Native Buildpacks, the Paketo project should
+support adding bill of materials metadata across the different components that
+go into an app image. This RFC serves as an intent to implement the BOM and
+defines the minimal set of key/values to be included.
+
+## Motivation
+
+A BOM can include an array of different metadata about application dependencies
+and buildpack dependencies such as SHAs, URIs, versions, license information,
+vulnerability identifiers, etc. This information is valuable for users who want
+an accurate inventory of everything in their final application image, or that
+may have been used to build it. Having access to that information makes it
+easier to audit for vulnerabilities and license information.
+
+The Paketo Java Buildpacks as well as a handful of other Paketo Buildpacks
+already make the BOM available. Another motivation for doing this work is to
+achieve a consistent user experience across the Paketo project.
+
+## Detailed Explanation
+
+The following Paketo components should have BOM metadata attached to them:
+
+### Stacks
+Stacks (such as those found in the [Paketo Stacks repository](https://github.com/paketo-buildpacks/stacks))
+should have BOM metadata that includes in-depth information on all of the OS
+level packages installed as part of the stack.
+
+The minimal set of metadata for stacks should be:
+```
+{
```

Comment:
In my mind it would make sense to use the same schema for the stack BOM as we use for the regular BOM. Eventually CNB should provide a way to combine these into a single BOM and using the same schema in both places will set us up nicely for that future. We may choose to require certain fields be set in certain cases in our buildpacks (e.g. we will set "source.upstreamVersion" for every package entry but not for language modules), but it would be nice if the schema itself was consistent.

Reply:
I agree with this.

Reply:
addressed in https://github.com/paketo-buildpacks/rfcs/pull/57/commits/077a268c0465f665bff252635d084806264349c9